internal/storage/rocksdb/store.go

```diff
@@ -389,37 +423,67 @@ func (rdb *rocksDB) generateSST(snap *gorocksdb.Snapshot, cf *gorocksdb.ColumnFa
 	return os.Open(fileName)
 }
 
+type checkPointSnapshot struct {
+	tar *utils.StreamingTar
+	dir string
+}
+
+func (r *checkPointSnapshot) Read(p []byte) (n int, err error) {
+	return r.tar.Read(p)
+}
+
+func (r *checkPointSnapshot) Close() error {
+	r.tar.Close()
+	return os.RemoveAll(r.dir)
+}
+
 func (rdb *rocksDB) GetSnapshot() (io.ReadCloser, error) {
 	defer rdb.opts.statsCli.Timing("rocksdb.snapshot.get.latency.ms", time.Now())
-	snap := rdb.db.NewSnapshot()
-	defer rdb.db.ReleaseSnapshot(snap)
 
 	sstDir, err := storage.CreateTempFolder(rdb.opts.sstDirectory, sstPrefix)
 	if err != nil {
 		rdb.opts.lgr.Error("GetSnapshot: Failed to create temporary dir", zap.Error(err))
 		return nil, err
 	}
-	defer os.RemoveAll(sstDir)
 
-	normalSSTFile, err := rdb.generateSST(snap, rdb.normalCF, sstDir)
+	checkpoint, err := rdb.db.NewCheckpoint()
```

Comment:
Better to obtain the global mutation lock since this function interacts directly with the filesystem.